packages/core/src/api/interactions.ts

```diff
@@ -32,65 +51,166 @@ export class InteractionsAPI {
 	public async reply(
 		interactionId: Snowflake,
 		interactionToken: string,
-		{ files, ...data }: APIInteractionResponseCallbackData & { files?: RawFile[] },
+		body: APIInteractionResponseCallbackData &
+			RESTPostAPIInteractionCallbackQuery & { files?: RawFile[]; with_response?: false },
+		options?: Pick<RequestData, 'signal'>,
+	): Promise<void>;
+
+	/**
+	 * Replies to an interaction
+	 *
+	 * @see {@link https://discord.com/developers/docs/interactions/receiving-and-responding#create-interaction-response}
+	 * @param interactionId - The id of the interaction
+	 * @param interactionToken - The token of the interaction
+	 * @param body - The callback data for replying
+	 * @param options - The options for replying
+	 */
+	public async reply(
+		interactionId: Snowflake,
+		interactionToken: string,
+		{
+			files,
+			with_response,
+			...data
+		}: APIInteractionResponseCallbackData & RESTPostAPIInteractionCallbackQuery & { files?: RawFile[] },
 		{ signal }: Pick<RequestData, 'signal'> = {},
 	) {
-		await this.rest.post(Routes.interactionCallback(interactionId, interactionToken), {
-			files,
-			auth: false,
-			body: {
-				type: InteractionResponseType.ChannelMessageWithSource,
-				data,
-			},
-			signal,
-		});
+		return with_response
+			? this.rest.post(Routes.interactionCallback(interactionId, interactionToken), {
```

Comment:
If `with_response` isn't a truthy value, no API call will be done. Same for all the other methods 😅

Reply:
Wow, good catch. Not sure what I was doing here.